.travis.yml

```diff
@@ -136,6 +136,7 @@ matrix:
       install:
       before_script:
       script:
+        - git fetch --deepen 500 #this implies it only works with PR less than 500 commits in it
```

Comment:
There's
```
git:
  depth: N
```
in Travis. It looks a bit nicer, what do you think?

Reply:
yeah, i just used this to test locally the fix. i'll change as i do not mind either way.

Reply:
well actually I'll use `git.depth: false` that won't have the limit for 500 commit per pr (although still hope it won't come to this ever)

Reply:
`git.depth false` might cause timeouts, in this case, we should triple-check Travis in the busiest times.

Reply:
now it took 4-5 sec instead of 1-2 (when depth=50), why do you thing it is going to cause any issue ?

Reply:
That's nice, I didn't know this setting can be set for only one job. I've read a few timeout issues when depth false was applied, but 4-5 secs is nothing.